src/uct/ib/rdmacm/rdmacm_ep.h

```diff
@@ -8,13 +8,23 @@
 
 #include "rdmacm_iface.h"
 
+
+typedef struct uct_rdmacm_ep_op uct_rdmacm_ep_op_t;
+
+struct uct_rdmacm_ep_op {
+    ucs_queue_elem_t    queue_elem;
+    uct_completion_t    *user_comp;
+};
+
+
 struct uct_rdmacm_ep {
     uct_base_ep_t                      super;
     uct_sockaddr_priv_pack_callback_t  pack_cb;
     void                               *pack_cb_arg;
     uint32_t                           pack_cb_flags;
     int                                is_on_pending;
-    ucs_status_t                       status;     /* client error handling status */
+    ucs_queue_head_t                   ops;
+    ucs_status_t                       status;     /* client EP status */
```

Comment:
maybe instead of "status", update is_on_pending to contain flags, and add another flag like "UCT_RDMACM_EP_FLAG_ESTABLISHED"

Reply:
status is used for error handling flow too, uct_rdmacm_ep is not used on fast path to care about it's size complicating the logic with flags.

Reply:
where is status being set in case of error?

Reply:
https://github.com/openucx/ucx/blob/a6a6d4c1c46fe6125a0d876377d13088dd8f9e7b/src/uct/ib/rdmacm/rdmacm_ep.c#L210